app-backend/db/src/main/scala/filters/Filters.scala

```diff
@@ -152,4 +154,47 @@ object Filters {
       requestTypeF
     )
   }
+
+  def taskQP(taskQP: TaskQueryParameters)(
+      implicit putTaskStatus: Put[TaskStatus],
+      putGeom: Put[Projected[Polygon]]): List[Option[Fragment]] =
+    List(
+      taskQP.projectId map { qp =>
+        fr"project_id = $qp"
+      },
+      taskQP.layerId map { qp =>
+        fr"project_layer_id = $qp "
+      },
+      taskQP.status map { qp =>
+        fr"status = $qp "
+      },
+      taskQP.locked map {
+        case true  => fr"locked_by IS NOT NULL"
+        case false => fr"locked_by IS NULL"
+      },
+      taskQP.lockedBy map { qp =>
+        fr"locked_by = $qp "
+      },
+      taskQP.bboxPolygon match {
+        case Some(bboxPolygons) =>
+          val fragments = bboxPolygons.map(
+            bbox =>
+              fr"(_ST_Intersects(geometry, ${bbox}) AND geometry && ${bbox})"
+          )
+          Some(fr"(" ++ Fragments.or(fragments: _*) ++ fr")")
+        case _ => None
+      },
+      taskQP.actionUser map { qp =>
+        fr"task_actions.user_id = $qp"
+      },
+      taskQP.actionType map { qp =>
+        fr"task_actions.to_status = $qp"
```

Comment:
I guess it is because I am not very sure about the use case of the `actionType` QP, but could you help me understand why we want to apply it on the `to_status` field?